scipy/stats/tests/test_stats.py

```diff
@@ -2753,82 +2753,94 @@ def test_1samp_alternative(self):
-        assert_equal(pcos(np.arange(10) + 1, 4, kind=kind), result)
-
-    # multiple - 2
-    for (kind, result) in [('rank', 45.0),
-                           ('strict', 30.0),
-                           ('weak', 50.0),
-                           ('mean', 40.0)]:
-        assert_equal(pcos([1,2,3,4,4,5,6,7,8,9], 4, kind=kind), result)
-
-    # multiple - 3
-    assert_equal(pcos([1,2,3,4,4,4,5,6,7,8], 4), 50.0)
-    for (kind, result) in [('rank', 50.0),
-                           ('mean', 45.0),
-                           ('strict', 30.0),
-                           ('weak', 60.0)]:
-
-        assert_equal(pcos([1,2,3,4,4,4,5,6,7,8], 4, kind=kind), result)
-
-    # missing
-    for kind in ('rank', 'mean', 'strict', 'weak'):
-        assert_equal(pcos([1,2,3,5,6,7,8,9,10,11], 4, kind=kind), 30)
-
-    # larger numbers
-    for (kind, result) in [('mean', 35.0),
-                           ('strict', 30.0),
-                           ('weak', 40.0)]:
-        assert_equal(
-              pcos([10, 20, 30, 40, 50, 60, 70, 80, 90, 100], 40,
-                   kind=kind), result)
-
-    for (kind, result) in [('mean', 45.0),
-                           ('strict', 30.0),
-                           ('weak', 60.0)]:
-        assert_equal(
-              pcos([10, 20, 30, 40, 40, 40, 50, 60, 70, 80],
-                   40, kind=kind), result)
-
-    for kind in ('rank', 'mean', 'strict', 'weak'):
-        assert_equal(
-              pcos([10, 20, 30, 50, 60, 70, 80, 90, 100, 110],
-                   40, kind=kind), 30.0)
-
-    # boundaries
-    for (kind, result) in [('rank', 10.0),
-                           ('mean', 5.0),
-                           ('strict', 0.0),
-                           ('weak', 10.0)]:
-        assert_equal(
-              pcos([10, 20, 30, 50, 60, 70, 80, 90, 100, 110],
-                   10, kind=kind), result)
 
-    for (kind, result) in [('rank', 100.0),
-                           ('mean', 95.0),
-                           ('strict', 90.0),
-                           ('weak', 100.0)]:
-        assert_equal(
-              pcos([10, 20, 30, 50, 60, 70, 80, 90, 100, 110],
-                   110, kind=kind), result)
-
-    # out of bounds
-    for (kind, score, result) in [('rank', 200, 100.0),
-                                  ('mean', 200, 100.0),
-                                  ('mean', 0, 0.0)]:
-        assert_equal(
-              pcos([10, 20, 30, 50, 60, 70, 80, 90, 100, 110],
-                   score, kind=kind), result)
-
-    assert_raises(ValueError, pcos, [1, 2, 3, 3, 4], 3, kind='unrecognized')
+class TestPercentileOfScore:
+
+    def f(self, *args, **kwargs):
+        return stats.percentileofscore(*args, **kwargs)
+
+    @pytest.mark.parametrize("kind, result", [("rank", 40),
+                                              ("mean", 35),
+                                              ("strict", 30),
+                                              ("weak", 40)])
```

Comment:
```suggestion
    cases = [("rank", 40), ("mean", 35), ("strict", 30), ("weak", 40)]
    @pytest.mark.parametrize("kind, result", cases)
```
Just a thought.

Reply:
This does not make so much sense for the other cases (which have "longer" input) so I think I should stick with the current uniformity

Reply:
I suppose in this case I'd break the lines in the variable assignment, e.g.
```python3
cases = [("propagate", [], 1, np.nan),
         ("propagate", [np.nan], 1, np.nan),
         ("propagate", [np.nan], [0, 1, 2], [np.nan, np.nan, np.nan]),
         ("propagate", [1, 2], [1, 2, np.nan], [50, 100, np.nan]),
         ("omit", [1, 2, np.nan], [0, 1, 2], [0, 50, 100]),
         ("omit", [np.nan, np.nan], [0, 1, 2], [np.nan, np.nan, np.nan])])
@pytest.mark.parametrize("policy, a, score, result", cases)
rather than deep in the `parametrize` call, the principle being to break the lines in the simpler statement if possible. But it's a personal preference, so I'll resolve. (Please leave comments open until they're resolved by the reviewer, so that the reviewer can keep track.)

Reply:
Actually these later lines are breaking PEP8 standards for line length, so this would be a good solution. Unresolving.

Reply:
Should be fixed in 080d26c

> Please leave comments open until they're resolved by the reviewer, so that the reviewer can keep track.

Ah ok, I didn't know that was the principle. I thought since GH gives me (who is not a reviewer) the option to "resolve" then I should use it when reaching a decision (with some confidence).

Reply:
Some reviewers will mind (like me), but not all. Then again, some reviewers also leave their own comments open indefinitely, even after receiving adequate response. If I agree with the response, I try to resolve them immediately so they don't clutter things. If I don't, I'm glad it's still open so we can discuss it further.

I'll resolve this as the rest of the changes to `test_stats.py` look OK as far as PEP8 is concerned (except for blank lines needed before the functions, which I noted separately).